lib/domain/services/collection_services.dart

```diff
@@ -1,15 +1,69 @@
 import 'package:memo/data/repositories/collection_repository.dart';
+import 'package:memo/data/repositories/memo_repository.dart';
+import 'package:memo/domain/isolated_services/memory_stability_services.dart';
 import 'package:memo/domain/models/collection.dart';
+import 'package:memo/domain/transients/collection_status.dart';
 
+/// Handles all domain-specific operations pertaining to one or multiple [Collection]
 abstract class CollectionServices {
+  /// Retrieves all available [CollectionStatus] and keeps listening to any changes made to them
+  Future<List<Stream<CollectionStatus>>> listenToAllCollectionsStatus();
+
+  /// Retrieves all available [Collection]
   Future<List<Collection>> getAllCollections();
+
+  /// Retrieves the [Collection] (with [collectionId]) current memory stability
+  ///
+  /// This value ranges from `0` to `1`, meaning lesser the value, worse is the memory stability.
+  ///
+  /// If [Collection.isPristine], this will return `0`.
+  Future<double> getCollectionMemoryStability({required String collectionId});
 }
 
 class CollectionServicesImpl implements CollectionServices {
-  CollectionServicesImpl(this.repo);
+  CollectionServicesImpl({required this.collectionRepo, required this.memoRepo, required this.memoryServices});
+
+  final CollectionRepository collectionRepo;
+  final MemoRepository memoRepo;
+
+  final MemoryStabilityServices memoryServices;
+
+  @override
+  Future<List<Stream<CollectionStatus>>> listenToAllCollectionsStatus() async {
+    final collectionsStream = await collectionRepo.listenToAllCollections();
+    return collectionsStream
+        .map(
+          // Asynchronously transform the stream due to the async calculations
+          (collectionStream) => collectionStream.asyncMap((collection) async {
+            double? memoryStability;
+            if (collection.isCompleted) {
+              memoryStability = await _getMemosAverageMemoryStability(collectionId: collection.id);
+            }
 
-  final CollectionRepository repo;
+            return CollectionStatus(collection, memoryStability);
+          }),
+        )
+        .toList();
+  }
 
   @override
-  Future<List<Collection>> getAllCollections() => repo.getAllCollections();
+  Future<List<Collection>> getAllCollections() => collectionRepo.getAllCollections();
+
+  @override
+  Future<double> getCollectionMemoryStability({required String collectionId}) =>
+      _getMemosAverageMemoryStability(collectionId: collectionId);
+
+  Future<double> _getMemosAverageMemoryStability({required String collectionId}) async {
+    final allCollectionMemos = await memoRepo.getAllMemos(collectionId: collectionId);
+
+    var averageStability = 0.0;
+    allCollectionMemos.forEach((memo) {
+      final stability = memoryServices.evaluateMemoryStability(memo);
+      if (stability != null) {
+        averageStability += stability;
+      }
+    });
+
+    return averageStability / allCollectionMemos.length;
```

Comment:
I think that the discussion that we had some days ago in Discord about counting the `pristine` memos in the average math could be summarized here

Reply:
Actually, counting only pristine versus all memos are equally bad, because they should never be used in the application unless the collection has all memos executed, so, even though this still returns a value, it may not  mean much, given the fact that only fully completed collections should be evaluated

Reply:
I was remembering now from our discussion about only show the memory stability when all collection cards were finished. With this definition in mind, having memos in this function without stability isn't an inconsistency? In theory, all of them should have a stability at this function